src/main/java/spoon/support/reflect/declaration/CtRecordImpl.java

```diff
@@ -100,12 +104,20 @@ public void accept(CtVisitor v) {
 	public <C extends CtType<Object>> C addTypeMemberAt(int position, CtTypeMember member) {
 		// a record can have only implicit instance fields and this is the best point to preserve the invariant
 		// because there are multiple ways to add a field to a record
+		String memberName = member.getSimpleName();
+
 		if (member instanceof CtField && !member.isStatic()) {
+			// We add the field in addRecordComponent. Afterward, however, JDT visits the Field itself -> Duplication.
```

Comment:
This feels weird here, as we drag JDT specific code into unrelated places. Can we avoid the duplication somewhere else?

Reply:
If you have an idea, feel free. I didn't find anything nice.

Reply:
I moved it to the `ParentExiter`.